src/libraries/System.Private.CoreLib/src/System/IO/File.cs

```diff
@@ -52,6 +52,51 @@ public static StreamWriter AppendText(string path)
             return new StreamWriter(path, append: true);
         }
 
+#if NET6_0_OR_GREATER
```

Comment:
using `MS_IO_REDIST` to determine that would be more clear about the intentions

But it would be still far from perfect to have so many `#if`s here.

Please make `File` a `partial` class and move all the new methods to `File.netcoreapp.cs` file and include it in the `Shared` project:

https://github.com/dotnet/runtime/blob/3846a9c4e5529ab1da1d58058f86502da85fead6/src/libraries/System.Private.CoreLib/src/System.Private.CoreLib.Shared.projitems#L409

Reply:
Done.